modules/core/src/viewports/viewport.ts

```diff
@@ -234,22 +239,24 @@ export default class Viewport {
   /**
    * Projects xyz (possibly latitude and longitude) to pixel coordinates in window
    * using viewport projection parameters
-   * - [longitude, latitude] to [x, y]
    * - [longitude, latitude, Z] => [x, y, z]
    * Note: By default, returns top-left coordinates for canvas/SVG type render
    *
-   * @param {Array} lngLatZ - [lng, lat] or [lng, lat, Z]
+   * @param {Array} xyz - [lng, lat, Z]
    * @param {Object} opts - options
    * @param {Object} opts.topLeft=true - Whether projected coords are top left
    * @return {Array} - [x, y] or [x, y, z] in top left coords
    */
-  project(xyz: number[], {topLeft = true}: {topLeft?: boolean} = {}): number[] {
+  project(
```

Comment:
Can you try using function overloads? https://github.com/visgl/deck.gl/pull/9313#discussion_r1908341371